src/EnergyPlus/DaylightingManager.cc

```diff
@@ -4766,38 +4545,18 @@ void GetDaylightingControls(EnergyPlusData &state, bool &ErrorsFound)
                                      ipsc->cAlphaFieldNames(NumAlpha)));
             ErrorsFound = true;
         }
-        daylightControl.DaylRefPtNum.allocate(curTotalDaylRefPts);
-        daylightControl.FracZoneDaylit.allocate(curTotalDaylRefPts);
-        daylightControl.IllumSetPoint.allocate(curTotalDaylRefPts);
-        daylightControl.DaylIllumAtRefPt.allocate(curTotalDaylRefPts);
-        daylightControl.GlareIndexAtRefPt.allocate(curTotalDaylRefPts);
-        daylightControl.DaylRefPtAbsCoord.allocate(curTotalDaylRefPts);
-        daylightControl.DaylRefPtInBounds.allocate(curTotalDaylRefPts);
-        daylightControl.RefPtPowerReductionFactor.allocate(curTotalDaylRefPts);
-        daylightControl.BacLum.allocate(curTotalDaylRefPts);
-        daylightControl.TimeExceedingGlareIndexSPAtRefPt.allocate(curTotalDaylRefPts);
-        daylightControl.TimeExceedingDaylightIlluminanceSPAtRefPt.allocate(curTotalDaylRefPts);
-
-        for (int refPt = 1; refPt <= curTotalDaylRefPts; ++refPt) {
-            daylightControl.DaylRefPtNum(refPt) = 0;
-            daylightControl.FracZoneDaylit(refPt) = 0.0;
-            daylightControl.IllumSetPoint(refPt) = 0.0;
-            daylightControl.DaylIllumAtRefPt(refPt) = 0.0;
-            daylightControl.GlareIndexAtRefPt(refPt) = 0.0;
-            daylightControl.DaylRefPtInBounds(refPt) = true;
-            daylightControl.RefPtPowerReductionFactor(refPt) = 1.0;
-            daylightControl.BacLum(refPt) = 0.0;
-            daylightControl.TimeExceedingGlareIndexSPAtRefPt(refPt) = 0.0;
-            daylightControl.TimeExceedingDaylightIlluminanceSPAtRefPt(refPt) = 0.0;
-            daylightControl.DaylRefPtAbsCoord(refPt) = {0.0, 0.0, 0.0};
+
+        daylightControl.refPts.allocate(curTotalDaylRefPts);
+
+        for (auto &refPt : daylightControl.refPts) {
+            refPt = DaylRefPt();
```

Comment:
Could (and probably should) have used `new (&refPt) DaylRefPt();` (in-place constructor).